openstudiocore/src/model/test/SiteGroundTemperatureBuildingSurface_GTest.cpp

```diff
@@ -0,0 +1,239 @@
+  ASSERT_FALSE(groundTemp.isSeptemberGroundTemperatureDefaulted());
+  ASSERT_EQ(23.0, groundTemp.septemberGroundTemperature());
+  ASSERT_EQ(23.0, groundTemp.getTemperatureByMonth(9));
+  ASSERT_FALSE(groundTemp.isOctoberGroundTemperatureDefaulted());
+  ASSERT_EQ(22.0, groundTemp.octoberGroundTemperature());
+  ASSERT_EQ(22.0, groundTemp.getTemperatureByMonth(10));
+  ASSERT_FALSE(groundTemp.isNovemberGroundTemperatureDefaulted());
+  ASSERT_EQ(21.0, groundTemp.novemberGroundTemperature());
+  ASSERT_EQ(21.0, groundTemp.getTemperatureByMonth(11));
+  ASSERT_FALSE(groundTemp.isDecemberGroundTemperatureDefaulted());
+  ASSERT_EQ(20.0, groundTemp.decemberGroundTemperature());
+  ASSERT_EQ(20.0, groundTemp.getTemperatureByMonth(12));
+
+  // Reset them all and make sure they're all defaulted again.
+  groundTemp.resetAllMonths();
+  for (int i = 1; i <= 12; ++i) {
+    ASSERT_TRUE(groundTemp.isMonthDefaulted(i));
+  }
+
+  // Test the setTemperatureByMonth and getTemperatureByMonth methods.
+  std::vector<double> temperatures;
+  temperatures.reserve(12);
+  for (int i=0; i < 12; ++i) {
+    temperatures.push_back(i + 18.0);
+  }
+  ASSERT_EQ(temperatures.size(), 12);
+  for (int i=1; i <= 12; ++i) {
+    groundTemp.setTemperatureByMonth(i, temperatures[i-1]);
+    ASSERT_FALSE(groundTemp.isMonthDefaulted(i));
+    ASSERT_EQ(temperatures[i-1], groundTemp.getTemperatureByMonth(i));
+  }
+
+  // Reset them one by one with the resetTemperatureByMonth method and check that it worked.
+  for (int i = 1; i <= 12; ++i) {
+    groundTemp.resetTemperatureByMonth(i);
+    ASSERT_TRUE(groundTemp.isMonthDefaulted(i));
+  }
+
+  // Test the vector getter/setters
+  bool setWorked = groundTemp.setAllMonthlyTemperatures(temperatures);
+  ASSERT_TRUE(setWorked);
+  std::vector<double> returned_temperatures = groundTemp.getAllMonthlyTemperatures();
+  ASSERT_EQ(temperatures.size(), 12);
+  for (int i=0; i < 12; ++i) {
+    ASSERT_EQ(temperatures[i], returned_temperatures[i]);
+    ASSERT_FALSE(groundTemp.isMonthDefaulted(i+1));
+    ASSERT_EQ(temperatures[i], groundTemp.getTemperatureByMonth(i+1));
+  }
+
+  // Try to set the monthly temperatures with a vector of the wrong length
+  temperatures.push_back(20.4);
+  ASSERT_EQ(temperatures.size(), 13);
+  setWorked = groundTemp.setAllMonthlyTemperatures(temperatures);
+  ASSERT_FALSE(setWorked);
+
+  // Test the ...byMonth getter/setters with MonthOfYear enums
+  groundTemp.resetAllMonths();
+  groundTemp.setTemperatureByMonth(openstudio::MonthOfYear::Jan, 15.0);
+  ASSERT_EQ(groundTemp.getTemperatureByMonth(openstudio::MonthOfYear::Jan), 15.0);
+  groundTemp.resetTemperatureByMonth(openstudio::MonthOfYear::Jan);
+  ASSERT_TRUE(groundTemp.isMonthDefaulted(openstudio::MonthOfYear::Jan));
+
+  // Test exceptions on an invalid month
+  try {
+    groundTemp.getTemperatureByMonth(13);
+  } catch (openstudio::Exception const & err) {
+    ASSERT_TRUE(boost::algorithm::ends_with(err.message(), "Invalid Month 13"));
```

Comment:
@nmerket Are you wanting to just make sure the function throws an exception? Or do you also want to make sure the error message ends with that string? If it is only the former then you can just call
`EXPECT_THROWS_ANY(groundTemp.getTemperatureByMonth(13));`

If it is the latter as well then I would encourage you to find a non-boost way to check the error message. We are trying to reduce the usages of Qt and Boost as much as possible.